examples/SharpBrick.PoweredUp.Examples/BaseExample.cs

```diff
@@ -83,11 +86,38 @@ public void InitHost(bool enableTrace)
 
                     if (enableTrace)
                     {
-                        builder.AddFilter("SharpBrick.PoweredUp.Bluetooth.BluetoothKernel", LogLevel.Debug);
+                        builder.AddFilter("SharpBrick.PoweredUp.Bluetooth.BluetoothKernel", LogLevel.Debug)
+                        //by this you can also disable the debug-output of the BlueGiga; all outputs of BlueGigaBLEPoweredUpBluetoothAdapater are made by LogDebug
+                        //so setting:
+                        //.AddFilter("SharpBrick.PoweredUp.BlueGigaBLE.BlueGigaBLEPoweredUpBluetoothAdapater", LogLevel.Information);
+                        //will also disable the output (but it is still send to the ILogger, which then filters)
+                        //you can also set the option TraceDebug to false in the AddBlueGigaBLEBluetooth below
+                        .AddFilter("SharpBrick.PoweredUp.BlueGigaBLE.BlueGigaBLEPoweredUpBluetoothAdapater", LogLevel.Debug);
                     }
-                })
-                .AddWinRTBluetooth()
-                ;
+                });
+            if (bluetoothImplementation == BluetoothImplementation.WinRT)
+            {
+                serviceCollection.AddWinRTBluetooth();
+            };
+
+            if (bluetoothImplementation == BluetoothImplementation.BlueGiga)
+            {
+                if (blueGigaBLEOptions == null)
```

Comment:
Code styling is really a personal issue - I'm now 35+ years in programming and have done so in a lot of languages (MASM/assembler, Basic, VisualBasic, Informix 4 GL, Pascal, Java, SQL, JavaScript (+HTML/CSS), C, C++ and C# just to name the main once). I will try to get Visual Studio in supporting me more on that (.editorconfig) and hopefully get the thinks done :-)

- I will remerge your main with .NET 5  first into my repo-branch
- It doesn't need a quick release. When do you appr. plan v4.0?
- define the interface for the bluetooth-stack-query. I will see what I can do for BlueGiga to implement that query. It's a serial port device (from OS point of view) and hopefully there is an OS-independent way to query some kind of device-identifier

Reply:
Regarding style guides for curly brackets following IF (as mentioned above): I cannot find any real style-guide on this. So every time there is just a single statement in the IF-case, I shouldn't use curly brackets? Or in other words: Use curly brackets after "if (condition)" when there are multiple statements to be done when the condition is true?
I've tried the standard .editorconfig for .NET (which shall fullfill the .NET style guide as mentioned by Microsoft) and this tells me I SHOULD use curly brackets even when only one statement has to be executed. Same for FOR-loops with only one statement...

Reply:
yeah. C# use for one-liners in a "block" curly brackets. I know. Was hard for me as well in 2002 ;)

Reply:
regards v4.0 date ... that is the thing ... i do not have a plan. So I probe when people want to have it so I can see how fast I need to go.

Realistically I would say in a month. There are 3-4 smaller feature thingies which are breaking and need to be done and the .NET 5 and C# 9.0 update.

Reply:
OK, this sounds googd.
So it would be the best for me also directly switch to C#9 so that you can take it into 4.0?

 (actually I have C# 8 explicitly set in Visual Studio by using
`<PropertyGroup>
    <LangVersion>8.0</LangVersion>
  </PropertyGroup>`
in the csproj-file fo the BlueGigaBLE-implementation (not yet pushed)

Reply:
yeah C# 9.0 branch is the latest and greatest.

Reply:
I did a lot of styling and refactoring in my commit/push from today. I also switched to C# 9.0 and used init instead of set in properties whereever possible and laso used pattern-matching where appicapable.
Not sure about if this PR is then still valid, because I did it as a PR against your main-branch at first...
Please comment what is the best way for you to handle... not so deep in GitHub/git, because mainly working with SVN in company....